### PR TITLE
CE-227 Bug Fix - IncompleteTransactionHistory error thrown by mongod

### PR DESCRIPTION
**Pull request by ayesharustgi-delphix**
<details open>
<summary><h2> Background </h2></summary>

CE-227 Bug Fix - IncompleteTransactionHistory error thrown by mongod

Took the strategy:
pause → fsync → snapshot → resume
</details>

<details open>
<summary><h2> Problem </h2></summary>

Provide a clear description of the high-level problem you are trying to
solve. The problem statement should be written in terms of a specific
symptom that affects users or the business. The problem statement should
not be written in terms of the solution. If possible, include a minimal
reproducible example (MRE) with steps to reproduce, expected results,
and actual results.
</details>

<details open>
<summary><h2> Solution </h2></summary>

Provide a clear description of the high-level solution you have chosen.
If there were other possible solutions that you considered and rejected,
mention those along with the corresponding reasoning. Do not describe
implementation details when writing about the solution; these should go
into the implementation section instead.
</details>

<details>
<summary><h2> Testing Done </h2></summary>

Provide a clear description of how this change was tested. At minimum
this should include proof that a computer has executed the changed
lines. Ideally this should include an automated test or an explanation
as to why this pull request has no tests.
</details>

### CODE REVIEW

File: src/mongopy.py [hunk 1]
```diff
@@ -720,6 +734,12 @@ def staged_post_snapshot(
 
     common.add_debug_heading_block("End Staged Post Snapshot")
     # ADD start Balancer
+    if staged_source.parameters.enable_clustersync:
+        mongosync_obj = MongoSync(staged_source=staged_source,
+                                  repository=repository,
+                                  check_params=False)
+        mongosync_obj.resume_sync()
```
Comment (manaskhandeshe-delphix) [inline src/mongopy.py:741]: Let's wait for say 30 seconds, verify if RESUME state is seen.

**Review by manaskhandeshe-delphix — CHANGES_REQUESTED** [review]

File: src/mongopy.py [hunk 2]
```diff
@@ -479,6 +480,19 @@ def staged_pre_snapshot(
                     and progress_json["progress"]["state"] == "RUNNING"
                 ):
                     ret = 1
+                    mongosync_obj.pause_sync()
```
[the remark this answers is not in the document]
Comment (ayesharustgi-delphix) [reply src/mongopy.py:489]: Done!

File: src/mongopy.py [hunk 3]
```diff
@@ -479,6 +480,19 @@ def staged_pre_snapshot(
...
+                    time.sleep(30)
+                    resource_obj = Resource(
+                        connection=staged_source.staged_connection,
+                        hidden_directory="")
+                    mongo_obj = MongoDB(repository, resource_obj)
+                    mongo_obj.fsyncLock(
```
[the remark this answers is not in the document]
Comment (ayesharustgi-delphix) [reply src/mongopy.py]: Done!

File: src/mongopy.py (hunk 1 above)
Comment (ayesharustgi-delphix) [reply src/mongopy.py]: Done!

**Review by manaskhandeshe-delphix — APPROVED** [review]

**Pull request merged**